src/app/beer_garden/events/processors.py

```diff
@@ -125,16 +128,47 @@ def register(self, processor, manage: bool = True):
 class HttpEventProcessor(QueueListener):
     """Publish events using an EasyClient"""
 
-    def __init__(self, easy_client=None, black_list=None, **kwargs):
+    def __init__(
+        self, easy_client=None, black_list=None, reconnect_action=None, **kwargs
+    ):
         super().__init__(**kwargs)
 
         self._ez_client = easy_client
         self._black_list = black_list or []
+        self._reconnect_action = reconnect_action
 
     def process(self, event: Event):
+
         try:
             if event.name not in self._black_list:
                 event.garden = beer_garden.config.get("garden.name")
                 self._ez_client.publish_event(event)
         except Exception as ex:
             logger.exception(f"Error publishing EasyClient event: {ex}")
+
+            self.reconnect()
+
+    def reconnect(self):
+
+        # Mark not processing and stop accepting events
+        self._processing = False
+
+        # Purge the current Queue
+        while not self._queue.empty():
+            self._queue.get()
+
+        # Back-off connection from EZ Client
+        wait_time = 0.1
+        while not self.stopped() and not self._processing:
+            if self._ez_client.can_connect():
+                self._processing = True
```

Comment:
Should add an warning-level log message indicating that we've successfully reconnected and are exiting this reconnect state.

Even though this feels like an info-level I'd say it's better to match the level of when we entered the reconnect state. That way if the logging level is set to warning it won't get lost.